services/server/src/server/services/workers/verificationWorker.ts

```diff
@@ -256,6 +267,219 @@ async function _verifyFromEtherscan({
+    return {
+      errorExport: {
+        customCode: "cannot_fetch_bytecode",
+        errorId: uuidv4(),
+      },
+    };
+  }
+
+  if (runtimeBytecode === "0x") {
+    return {
+      errorExport: {
+        customCode: "contract_not_deployed",
+        errorId: uuidv4(),
+      },
+    };
+  }
+
+  // Fetch creation data to be used in the SourcifyChainMock
+  let creationData: {
+    creationBytecode?: string;
+    deployer?: string;
+    blockNumber?: number;
+    txIndex?: number;
+  } = {};
+  if (creatorTxHash) {
+    try {
+      const creatorTx = await sourcifyChain.getTx(creatorTxHash);
+      const { creationBytecode, txReceipt } =
+        await sourcifyChain.getContractCreationBytecodeAndReceipt(
+          checksumAddress,
+          creatorTxHash,
+          creatorTx,
+        );
+      creationData = {
+        creationBytecode,
+        deployer: creatorTx.from,
+        blockNumber: creatorTx.blockNumber ?? undefined,
+        txIndex: txReceipt.index ?? undefined,
+      };
+    } catch (error: any) {
+      logger.debug(
+        "Failed to fetch creation data for similarity verification",
+        {
+          chainId,
+          address: checksumAddress,
+          creatorTxHash: creatorTxHash,
+          error: error?.message,
+        },
+      );
+    }
+  }
+
+  const mockChain = new SourcifyChainMock(
+    {
+      onchain_runtime_code: runtimeBytecode,
+      onchain_creation_code: creationData.creationBytecode,
+      deployer: creationData.deployer,
+      block_number: creationData.blockNumber,
+      transaction_index: creationData.txIndex,
+      transaction_hash: creatorTxHash,
+      address: checksumAddress,
+    },
+    Number(chainId),
+    checksumAddress,
+  );
+
+  for (const candidate of candidates) {
+    const compilation = createPreRunCompilationFromCandidate(candidate);
+    if (!compilation) {
+      continue;
+    }
+
+    const verification = new Verification(
+      compilation,
+      mockChain,
+      checksumAddress,
+      creatorTxHash,
+    );
+
+    try {
+      await verification.verify();
+    } catch (error: any) {
+      if (error instanceof SourcifyLibError) {
+        logger.debug("Similarity candidate verification failed", {
+          chainId,
+          address: checksumAddress,
+          error: error.code,
+        });
+        continue;
+      }
+
+      return {
+        errorExport: {
+          customCode: "internal_error",
+          errorId: uuidv4(),
+        },
+      };
+    }
+
+    const { runtimeMatch, creationMatch } = verification.status;
+
+    if (runtimeMatch !== null || creationMatch !== null) {
+      logger.info("Similarity verification matched candidate", {
+        chainId,
+        address: checksumAddress,
+      });
+      return {
+        verificationExport: verification.export(),
+      };
+    }
+  }
+
+  return {
+    errorExport: {
+      customCode: "no_similar_match_found",
+      errorId: uuidv4(),
+    },
```

Comment:
Should we include the `onchainRuntimeCode`, `onchainCreationCode` and `creationTransactionHash` here?

Reply:
For debugging purposes. Same reasoning why we store it on the other verification endpoints.